lib/plugins/aws/deploy/compile/events/apiGateway/lib/apiKeys.js

```diff
@@ -6,41 +6,33 @@ const BbPromise = require('bluebird');
 module.exports = {
   compileApiKeys() {
     if (this.serverless.service.provider.apiKeys) {
-      if (this.serverless.service.provider.apiKeys.constructor !== Array) {
+      if (!Array.isArray(this.serverless.service.provider.apiKeys)) {
         throw new this.serverless.classes.Error('apiKeys property must be an array');
       }
 
-      let apiKeyNumber = 0;
-      this.serverless.service.provider.apiKeys.forEach(apiKey => {
-        apiKeyNumber++;
+      _.forEach(this.serverless.service.provider.apiKeys, (apiKey, i) => {
+        const apiKeyNumber = i + 1;
+
         if (typeof apiKey !== 'string') {
           throw new this.serverless.classes.Error('API Keys must be strings');
         }
 
-        const apiKeysTemplate = `
-        {
-          "Type" : "AWS::ApiGateway::ApiKey",
-          "Properties" : {
-            "Enabled" : true,
-            "Name" : "${apiKey}",
-            "StageKeys" : [{
-              "RestApiId": { "Ref": "ApiGatewayRestApi" },
-              "StageName": "${this.options.stage}"
-            }]
+        _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources, {
+          [`ApiGatewayApiKey${apiKeyNumber}`]: {
+            Type: 'AWS::ApiGateway::ApiKey',
+            Properties: {
+              Enabled: true,
+              Name: apiKey,
+              StageKeys: [{
+                RestApiId: { Ref: this.restApiLogicalId },
+                StageName: this.options.stage,
+              }],
+            },
+            DependsOn: this.apiGateWayDeploymentLogicalId,
```

Comment:
Could you rename this from `apiGateWayDeploymentLogicalId` to `apiGatewayDeploymentLogicalId` so that we have an overall consistent naming schema?